### PR TITLE
I265 banner flipper

### PR DESCRIPTION
**Pull request by maxkadel**
Connected to #265 

Things I'm particularly interested in feedback on:
- Is it a problem that there is no auth on the api allowing you to turn the banner on / off?
- The HTML is escaped in the json - this seems safer, and then the allsearch frontend can unescape the html
- Is the documentation clear?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -88,3 +88,61 @@ the swagger.yaml file meets the OpenAPI standard.
...
+curl -X DELETE http://allsearch-api/flipper/api/features/banner/boolean
+```
+
+### Feature Flipper
+In order to create a new feature flipper for an environment:
+See [Flipper API documentation](https://www.flippercloud.io/docs/api#create-a-new-feature)
+
+```bash
+curl -X POST -d "name=banner" https://allsearch-api-staging.princeton.edu/flipper/api/features
+```
```
Comment (sandbergja) [inline README.md:148]: Thanks for these docs!

File: config/routes.rb [hunk 2]
```diff
@@ -20,4 +21,6 @@
   get '/search/pulmap', to: 'pulmap#show', defaults: { format: 'json' }
   get '/search/staff/', to: 'library_staff#show', defaults: { format: 'json' }
   get '/search/website', to: 'library_website#show', defaults: { format: 'json' }
+
+  mount Flipper::Api.app(Flipper) => '/flipper/api'
```
Comment (sandbergja) [inline config/routes.rb:25]: My understanding is that this gives anybody in the world the ability to turn the banner on and off.  I'd much rather limit this to us, either by

- keeping this api but putting it behind some kind of auth,
- requiring us to SSH into a box to access it (via curl, or maybe we just do it via the rails console),
- Maybe a cap task or something if we want to save the trouble of explicitly SSHing into a box,
- something else

File: lib/tasks/banner.rake [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
```
Comment (sandbergja) [inline lib/tasks/banner.rake:2]: These rake tasks are great!

File: spec/models/banner_spec.rb [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+  end
+
+  it 'only allows valid alert statuses' do
+    expect do
+      banner.alert_status = 'warning'
+      banner.save!
+    end.to change(banner, :alert_status)
+
+    expect do
+      banner.alert_status = 'potato'
```
Comment (sandbergja) [inline spec/models/banner_spec.rb:41]: 🥔

**Review by sandbergja — CHANGES_REQUESTED** [review]
Thanks, @maxkadel 🦄 !  This works really nicely.

To answer your question about auth on the flipper API, my vote is not to make that available without some kind of protections.  It seems like a pain to troubleshoot some random person turning the banner on and off unexpectedly, and could lead to users getting, say, outdated information (for example, if somebody turns on an outdated banner that says "The library is closed today"), or not getting the information they need (for example, if we turn on a banner with timely information, but somebody else turns it off).

File: config/routes.rb (hunk 2 above)
Comment (maxkadel) [reply config/routes.rb:25]: Yeah. I was trying to find some balance where someone like Beth could turn it on or off, but I think maybe the need for curl / post requests makes that unlikely.

I think maybe the cap task makes the most sense. Maybe we could even put it in an Ansible Tower playbook so you don't have to have your own environment for it.

Comment (sandbergja) [reply config/routes.rb:25]: I like the ansible tower idea a lot!  That would be really slick.

**Review by sandbergja — APPROVED** [review]
Thanks, @maxkadel 🦄 !

**Pull request merged**